src/util/messagepipe.h

```diff
@@ -1,109 +1,83 @@
 #pragma once
 
-#include <QScopedPointer>
-#include <QSharedPointer>
+#include <memory>
 
 #include "rigtorp/SPSCQueue.h"
-
 #include "util/class.h"
-#include "util/reference.h"
+
+template<class SenderMessageType, class ReceiverMessageType>
+class TwoWayMessagePipe;
 
 // MessagePipe represents one side of a TwoWayMessagePipe. The direction of the
 // pipe is with respect to the owner so sender and receiver are
 // perspective-dependent.
 template <class SenderMessageType, class ReceiverMessageType>
 class MessagePipe {
   public:
-    MessagePipe(rigtorp::SPSCQueue<SenderMessageType>& receiver_messages,
-                rigtorp::SPSCQueue<ReceiverMessageType>& sender_messages,
-                BaseReferenceHolder* pTwoWayMessagePipeReference)
+    MessagePipe(std::shared_ptr<rigtorp::SPSCQueue<SenderMessageType>> receiver_messages,
+            std::shared_ptr<rigtorp::SPSCQueue<ReceiverMessageType>> sender_messages)
             : m_receiver_messages(receiver_messages),
-              m_sender_messages(sender_messages),
-              m_pTwoWayMessagePipeReference(pTwoWayMessagePipeReference) {
+              m_sender_messages(sender_messages) {
```

Comment:
Would `std::move` here help with not having to copy `sender_messages` and `receiver_messages`